extensions/aws/tests/features/steps/s3_server_container.py

```diff
@@ -0,0 +1,62 @@
+import json
+import logging
+
+from minifi_test_framework.containers.container import Container
+from minifi_test_framework.core.helpers import wait_for_condition
+from minifi_test_framework.core.minifi_test_context import MinifiTestContext
+
+
+class S3ServerContainer(Container):
+    def __init__(self, test_context: MinifiTestContext):
+        super().__init__("adobe/s3mock:3.12.0", f"s3-server-{test_context.scenario_id}", test_context.network)
+        self.environment.append("initialBuckets=test_bucket")
+
+    def deploy(self):
+        super().deploy()
+        finished_str = "Started S3MockApplication"
+        return wait_for_condition(
+            condition=lambda: finished_str in self.get_logs(),
+            timeout_seconds=15,
+            bail_condition=lambda: self.exited,
+            context=None)
+
+    def check_kinesis_server_record_data(self, container_name, record_data):
```

Comment:
These checks used to have a `@retry_check()` annotation; is that no longer needed?